bigquery_etl/glean_usage/common.py

```diff
@@ -113,88 +141,120 @@ def _extract_dataset_from_glob(pattern):
-                output_only=args.output_only,
-            ),
-            baseline_tables,
+class GleanTable:
+    """Represents a generated Glean table."""
+
+    def __init__(self):
+        """Init Glean table."""
+        self.target_table_id = ""
+        self.prefix = ""
+        self.custom_render_kwargs = {}
+        self.no_init = True
+
+    def generate_per_app_id(
+        self,
+        project_id,
+        baseline_table,
+        output_dir=None,
+        output_only=False,
+    ):
+        """Generate the baseline table query per app_id."""
+        tables = table_names_from_baseline(baseline_table, include_project_id=False)
+
+        init_filename = f"{self.target_table_id}.init.sql"
+        query_filename = f"{self.target_table_id}.query.sql"
+        view_filename = f"{self.target_table_id[:-3]}.view.sql"
+        view_metadata_filename = f"{self.target_table_id[:-3]}.metadata.yaml"
+
+        table = tables[f"{self.prefix}_table"]
+        view = tables[f"{self.prefix}_view"]
+        render_kwargs = dict(
+            header="-- Generated via bigquery_etl.glean_usage\n",
+            project_id=project_id,
         )
-        if args.output_only:
-            return
-        logging.info(f"Dry runs successful for {len(baseline_tables)} table(s)")
-        # Now, actually run the queries.
-        if not args.dry_run:
-            pool.map(
-                partial(
-                    run_query_callback, args.project_id, date=args.date, dry_run=False
-                ),
-                baseline_tables,
+
+        render_kwargs.update(self.custom_render_kwargs)
+        render_kwargs.update(tables)
+
+        query_sql = render(query_filename, **render_kwargs)
+        view_sql = render(view_filename, **render_kwargs)
+        view_metadata = render(view_metadata_filename, format=False, **render_kwargs)
+
+        if not self.no_init:
+            try:
+                init_sql = render(init_filename, **render_kwargs)
+            except TemplateNotFound:
+                init_sql = render(query_filename, init=True, **render_kwargs)
+
+        if not (referenced_table_exists(view_sql)):
+            logging.info(
+                "Skipping view for table which doesn't exist:"
+                f" {self.target_table_id}"
             )
+            return
+
+        if output_dir:
+            write_sql(output_dir, view, "metadata.yaml", view_metadata)
+            write_sql(output_dir, view, "view.sql", view_sql)
+            write_sql(output_dir, table, "query.sql", query_sql)
+
+            if not self.no_init:
+                write_sql(output_dir, table, "init.sql", init_sql)
+
+            write_dataset_metadata(output_dir, view)
+
+        if output_only:
+            return
+
+        # dry run generated query
+        DryRun(
+            os.path.join(
+                output_dir, *list(baseline_table.split(".")[-2:]), query_filename
+            ),
+            query_sql,
+        ).is_valid()
```

Comment:
This doesn't actually do anything if the dry run fails, except for printing an error (like the old implementation). Not sure if we'd want to handle failed dry run cases?